src/libtriton/arch/arm/arm32/arm32Cpu.cpp

```diff
@@ -361,6 +407,37 @@ namespace triton {
                     throw triton::exceptions::Disassembly("Arm32Cpu::disassembly(): Invalid operand.");
                 } // switch
               } // for operand
+
+              /* NOTE: For some instructions the destination operand is
+               * optional (in which case the first source operand is used as
+               * destination). Capstone returns always all three operands of
+               * ARM instruction (i.e. make the destination operand explicit).
+               * However, it does not do the same for Thumb instruction. Here
+               * we make the destination operand explicit (in order to simplify
+               * the semantics implementation).
+               */
+              /* TODO (cnheitman): Discuss. Should we deal with this here or
+               * in the impementation of the semantic of each instruction?
+               * (The list of instructions will probably grow.)
```

Comment:
Yes we have to discuss about this, as you say the list will grow and it implies a very big overhead.

Reply:
I agree. For now, I'll proceed as in https://github.com/JonathanSalwan/Triton/pull/845#discussion_r374254838